layer/support/event_logging.h

```diff
@@ -136,6 +137,47 @@ using StringAttr = AttributeImpl<std::string, ValueType::kString>;
 using VectorInt64Attr =
     AttributeImpl<std::vector<int64_t>, ValueType::kVectorInt64>;
 
+// A compound event used by all Trace Event types. This class is shared among
+// all Trace Event types. Different event types specify their required
```

Comment:
nit: 'This class is shared among all Trace Event types. ' --> this seems redundant to me, I'd drop this sentence